torchaudio/csrc/lfilter.cpp

```diff
@@ -96,6 +98,12 @@ void lfilter_core_generic_loop(
   }
 }
 
+#ifndef USE_CUDA
+
+std::function<void(const torch::Tensor&, const torch::Tensor&, torch::Tensor&)>
+    cuda_lfilter_core_loop = lfilter_core_generic_loop;
```

Comment:
This is a cool trick, however, it feels like reducing the readability of code.
How about simply branching with USE_CUDA macro at L130 (inside of `else if (device.is_cuda())` branch?)


```c++
else if (device.is_cuda()) {
#ifdef USE_CUDA
    cuda_lfilter_core_loop(waveform, a_coeff_flipped, padded_output_waveform);
#else
    lfilter_core_generic_loop(waveform, a_coeff_flipped, padded_output_waveform);
#endif
}

```

Reply:
Good, it's more readable.